File: api/cron/random-message.ts
// api/cron/random-message.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';
import { bot } from '../../src/bot'; // Ajusta la ruta
import { sendRandomFunMessage } from '../../src/utils/schedulerHelpers'; // Ajusta la ruta

export default async function handler(
  request: VercelRequest,
  response: VercelResponse
) {
  // Opcional: Protección con secret
  // if (request.headers['x-cron-secret'] !== process.env.CRON_SECRET) {
  //   return response.status(401).send('Unauthorized');
  // }

  try {
    console.log("CRON JOB: Iniciando envío de mensaje aleatorio...");
    await sendRandomFunMessage(bot); // Pasa la instancia del bot
    console.log("CRON JOB: Envío de mensaje aleatorio completado.");
    response.status(200).send('Random message job executed successfully.');
  } catch (error: any) {
    console.error("CRON JOB: Error en el job de mensaje aleatorio:", error);
    response.status(500).send(`Error executing random message job: ${error.message}`);
  }
}

File: api/cron/random_message.py
# api/cron/random_message.py
import asyncio
import os
from dotenv import load_dotenv

project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
dotenv_path = os.path.join(project_root, '.env')
if os.path.exists(dotenv_path): load_dotenv(dotenv_path)

from app.bot_instance import bot
from app.utils.scheduler_helpers import trigger_random_fun_message

async def _main():
    print("Cron Job: Iniciando trigger_random_fun_message...")
    await trigger_random_fun_message(bot)
    await bot.session.close()
    print("Cron Job: trigger_random_fun_message completado.")

def handler(event, context):
    try:
        asyncio.run(_main())
        return { "statusCode": 200, "body": "Random message job executed." }
    except Exception as e:
        print(f"Error en el cron job random_message: {e}")
        return { "statusCode": 500, "body": f"Error: {e}" }